contributed_definitions/nyaml/NXdata_photoemission.yaml

```diff
@@ -0,0 +1,1143 @@
+  y(NX_FLOAT):
+    unit: NX_ANY
+    doc: |
+      This is an array holding the values to use for the y-axis of
+      data.  The units must be appropriate for the measurement.
+    dimensions:
+      rank: 1
+      dim: [[1, ny]]
+  z(NX_FLOAT):
+    unit: NX_ANY
+    doc: |
+      This is an array holding the values to use for the z-axis of
+      data.  The units must be appropriate for the measurement.
+    dimensions:
+      rank: 1
+      dim: [[1, nz]]
+  energy(NX_NUMBER):
+    unit: NX_ENERGY
+    doc: |
+      Calibrated energy axis.
+      
+      In an application definition, this could be a link to either
+      /entry/process/energy_calibration/calibrated_axis or
+      /entry/process/energy_correction/calibrated_axis.
+    \@type:
+      type: NX_CHAR
+      doc: |
+        The energy can be either stored as kinetic or as binding energy.
+      enumeration:
+        kinetic:
+          doc:
+          - |
+            Calibrated kinetic energy axis.
+          - |
+            xref:
+              spec: ISO 18115-1:2023
+              term: 3.35
+              url: https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:term:3.35
+        binding:
+          doc:
+          - |
+            Calibrated binding energy axis.
+          - |
+            xref:
+              spec: ISO 18115-1:2023
+              term: 12.16
+              url: https://www.iso.org/obp/ui/en/#iso:std:iso:18115:-1:ed-3:v1:en:term:12.16
+  \@energy_indices:
+  \@energy_depends:
+  momentum(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      Calibrated momentum axis.
+      Units are kg*m/s.
+  \@momentum_indices:
+  \@momentum_depends:
+  kx(NX_NUMBER):
+    unit: NX_WAVENUMBER
+    doc: |
+      Calibrated x axis in k-space.
+      Units are 1/Angström.
+  \@kx_indices:
+  \@kx_depends:
+  ky(NX_NUMBER):
+    unit: NX_WAVENUMBER
+    doc: |
+      Calibrated y axis in k-space.
+      Units are 1/Angström
+  \@ky_indices:
+  \@ky_depends:
+  kz(NX_NUMBER):
+    unit: NX_WAVENUMBER
+    doc: |
+      Calibrated z axis in k-space.
+      Units are 1/Angström.
+  \@kz_indices:
+  \@kz_depends:
+  angular0(NX_NUMBER):
+    unit: NX_ANGLE
+    doc: |
+      Fast-axis angular coordinate (or second slow axis if angularly integrated).
+  \@angular0_indices:
+  \@angular0_depends:
+  angular1(NX_NUMBER):
+    unit: NX_ANGLE
+    doc: |
+      Slow-axis angular coordinate (or second fast axis if simultaneously dispersed in
+      2 dimensions)
+  \@angular1_indices:
+  \@angular1_depends:
+  spatial0(NX_NUMBER):
+    unit: NX_LENGTH
+    doc: |
+      Fast-axis spatial coordinate (or second slow axis if spatially integrated)
+  \@spatial0_indices:
+  \@spatial0_depends:
+  spatial1(NX_NUMBER):
+    unit: NX_LENGTH
+    doc: |
+      Slow-axis spatial coordinate (or second fast axis if simultaneously dispersed in
+      2 dimensions)
+  \@spatial1_indices:
+  \@spatial1_depends:
+  delay(NX_NUMBER):
+    unit: NX_TIME
+    doc: |
+      Calibrated delay time.
```

Comment:
angularN/spatialN ?

Reply:
Yeah, sounds good, also more in line with the process base class. Do you want to do them same for the k-axes or just keep them as they are? After all, there's only kx, ky, kz, so the possible naming space is anyway quite limited.